test-framework/junit5/src/main/java/io/quarkus/test/junit/DisabledOnIntegrationTest.java

```diff
@@ -36,4 +36,18 @@
      * Reason for disabling this test
      */
     String value() default "";
+
+    /**
+     * The types of Quarkus application produced by the build for which this test is disabled.
+     * For example, if the tests are supposed to be disabled for built containers only,
+     * {@code ArtifactType.CONTAINER} would be set.
+     */
+    ArtifactType[] forArtifactTypes() default { ArtifactType.ALL };
+
+    enum ArtifactType {
+        ALL,
+        JAR,
+        CONTAINER,
+        NATIVE_BINARY,
```

Comment:
Would be nice to have `NATIVE_CONTAINER` + `JAR_CONTAINER` as well

Reply:
How come? Is there a reason you would want to run tests for jar container and not a native container?

Reply:
Maybe. Could think of a configuration/profile that uses an on-heap cache (e.g. caffeine) with an uber-jar, but no cache with a native image (because caches are not sooo great with serial GC in native images from GraalVM community).

Reply:
I would rather not do this unless we have a real need for ir

Reply:
Okay